src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,257 @@
+        properties = []
+        for col, dtype in df.dtypes.items():
+            name = f"{self._clean_column_name(col)}"
+            title = name.replace("_", " ").title()
+            if dtype == "object":
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            elif dtype == "int64":
+                prop = IntegerMetadataProperty(
+                    name=name, title=title, visible_for_annotators=self.visible_for_annotators
+                )
+            elif dtype == "float64":
+                prop = FloatMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            elif dtype == "bool":
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            else:
+                print(f"Unhandled data type for column {col}: {dtype}")
+                continue
+            properties.append(prop)
+        return properties
+
+    def _add_text_descriptives_to_metadata(
+        self, records: List[Union[FeedbackRecord, RemoteFeedbackRecord]], df: pd.DataFrame
+    ) -> List[Union[FeedbackRecord, RemoteFeedbackRecord]]:
+        """
+        Add the text descriptives metrics extracted previously as metadata
+        to a list of FeedbackDataset records.
+
+        Args:
+            records (List[Union[FeedbackRecord, RemoteFeedbackRecord]]): A list of FeedbackDataset or RemoteFeedbackDataset records.
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List[Union[FeedbackRecord, RemoteFeedbackRecord]]: A list of FeedbackDataset or RemoteFeedbackDataset records with extracted metrics added as metadata.
+        """
+        # Loop through the records and add the metrics to the metadata
+        modified_records = []
+        # Make tqdm optional
+        for record, metrics in tqdm(
+            zip(records, df.to_dict("records")),
+            total=len(records),
+            disable=not self.progress,
+            desc="Getting text descriptives...",
+        ):
+            record.metadata.update(metrics)
+            modified_records.append(record)
+        return modified_records
+
+    def update_dataset(
+        self, dataset: Union[FeedbackDataset, RemoteFeedbackDataset]
+    ) -> Union[FeedbackDataset, RemoteFeedbackDataset]:
+        """
+        Extract text descriptives metrics from a records in a FeedbackDataset
+        or RemoteFeedbackDataset, add them as metadata to the records and
+        return the updated dataset.
+
+        Args:
+            dataset (Union[FeedbackDataset, RemoteFeedbackDataset]): A FeedbackDataset or RemoteFeedbackDataset.
+
+        Returns:
+            Union[FeedbackDataset, RemoteFeedbackDataset]: A FeedbackDataset or RemoteFeedbackDataset with text descriptives metrics added as metadata.
+        """
+        # Extract records
+        records = dataset.records
```

Comment:
Here note that `dataset.records` is not a `List[FeedbackRecord]` for the `RemoteFeedbackDataset` but a `RemoteFeedbackRecords` class, so maybe it's worth doing something like the following?

```suggestion
        if isinstance(dataset, FeedbackDataset):
            records = dataset.records
        elif isinstance(dataset, RemoteFeedbackDataset):
            # Since `dataset` is a `RemoteFeedbackDataset` we need to fetch them all first, and then assign those to `records` as it's already a `List[FeedbackRecord]`
            records = dataset.pull().records
        else:
            raise ValueError(f"Provided `dataset` is of `type={type(dataset)}` while only `type=FeedbackDataset` or `type=RemoteFeedbackDataset` are allowed.")
```

Reply:
Great suggestion. Added it.